File: source/development/device-api/device-server-writing/writing_java/StepperMotorClass2.java
public void command_factory()
{
	String str = new String("DevReadPosition");
	command_list.addElement(new DevReadPositionCmd(str,
		 	     Tango_DEV_LONG,Tango_DEV_LONG,
			     "Motor number (0-7)",
			     "Motor position"));

	str =  new String("DevReadDirection");
	command_list.addElement(new TemplCommandInOut(str,
	                        "dev_read_direction",
				"direct_cmd_allowed"));
}

File: source/development/device-api/device-server-writing/writing/steppermotorclass2.cpp

void StepperMotorClass::command_factory()
{
	command_list.push_back(new DevReadPositionCmd("DevReadPosition",
						      Tango::DEV_LONG,
						      Tango::DEV_LONG,
						      "Motor number (0-7)",
						      "Motor position"));
						      
	command_list.push_back(
	    new TemplCommandInOut<Tango::DevLong,Tango::DevLong>
		((const char *)"DevReadDirection",
	         static_cast<Tango::Lg_CmdMethPtr_Lg>
			(&StepperMotor::dev_read_direction),
	         static_cast<Tango::StateMethPtr>
			(&StepperMotor::direct_cmd_allowed))
			      );
}

